addon/utils/cp/type-class.js

```diff
@@ -0,0 +1,15 @@
+import { computed } from '@ember/object';
+import { assert } from '@ember/debug';
+
+export default function typeClass(prefix) {
+  assert('You have to provide a prefix for typeClass', prefix);
+
+  return computed('type', function() {
+    let type = this.get('type') || 'default';
```

Comment:
We may want to add an assertion that `type` is a non-empty string.

Reply:
Why, if we have a default value here? (literally "default" 😉)

Reply:
I was talking about `type` and not `this.get('type')`. Since `type` is such a general term and current implementation hides the dependency, this would at least catch some issues (e.g. `this.get('type')` being a number, array or object). We could even assert that it's a supported string but I'm not sure if that wouldn't be too much.